green-hub/src/components/Articles/Articles.jsx

```diff
@@ -1,32 +1,44 @@
-import { useEffect, useState } from "react";
+import { useEffect, useState, useContext } from "react";
 import styles from "./Articles.module.css";
 import jsonData from "../../components/Articles/data.json";
 import heart from "../../../assets/images/page-articles/heart.png";
 import pinkHeart from "../../../assets/images/page-articles/pinkheart.png";
 import Navbar from "../Navbar/Navbar";
 import useAuth from "../../context/AuthContext";
 import Footer from "../Footer/Footer";
+import { db } from "../../config/firebase";
+import { doc, collection,setDoc,deleteDoc ,getDoc,query,getDocs,where} from "firebase/firestore";
+import {HeartsContext} from "../../context/HeartsContext";
 
 const Articles = () => {
   const [articleData, setArticleData] = useState(null);
-  const [clickedHearts, setClickedHearts] = useState([]);
+  
+  const {clickedHearts, setClickedHearts} = useContext(HeartsContext);
   const { currentUser } = useAuth();
 
   useEffect(() => {
     setArticleData(jsonData);
   }, []);
-
-  const handleHeartClick = (articleId) => {
-    if (clickedHearts.includes(articleId)) {
-      setClickedHearts(clickedHearts.filter((id) => id !== articleId));
-    } else {
+ 
+  
+  const handleHeartClick = async (articleId) => {
+      
+      const doc_id = `${articleId}_${currentUser.uid}`;
+      const docRef = doc(db,  "users_hearts", doc_id);
```

Comment:
trochę dziwnie, że to nie jest pod userem, tylko jest to osobna kolekcja "users_hearts". Zbędny krok.